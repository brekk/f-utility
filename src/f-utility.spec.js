import OLD from "f-utility"
import R from "ramda"
// import PKG from "../package.json"
import exam from "$build/tester"
/* const OLD = require("../old-f-utility") */

// exam("version", F => () => {
//   expect(F.version).toEqual(PKG.version)
// })
exam("snapshot", F => () => {
  const sortedSnapKeys = F.pipe(
    F.filter(F.is(Function)),
    F.keys,
    F.sort((a, b) => a - b)
  )(F)
  expect(sortedSnapKeys).toMatchSnapshot()
})
exam("shared interface", F => () => {
  const getMethods = F.pipe(F.filter(F.is(Function)), F.keys)
  expect(F.intersection(getMethods(F), getMethods(R))).toMatchSnapshot()
})

/* eslint-disable func-style */
exam("f-utility vs. ramda", F => () => {
  const futilityMethods = F.pipe(F.filter(F.is(Function)), F.keys)(F)
  const FUTILITY_VS_RAMDA = F.difference(futilityMethods, R.keys(R))
  expect(FUTILITY_VS_RAMDA).toMatchSnapshot()
})
exam("ramda vs. f-utility", F => () => {
  const futilityMethods = F.pipe(F.filter(F.is(Function)), F.keys)(F)
  const RAMDA_VS_FUTILITY = F.difference(R.keys(R), futilityMethods)
  expect(RAMDA_VS_FUTILITY).toMatchSnapshot()
})
exam("v3 vs. v4", F => () => {
  const futilityMethods = F.pipe(F.filter(F.is(Function)), F.keys)(F)
  const F3_VS_F4 = F.difference(R.keys(OLD), futilityMethods)
  expect(F3_VS_F4).toMatchSnapshot()
})
exam("v4 vs. v3", F => () => {
  const futilityMethods = F.pipe(F.filter(F.is(Function)), F.keys)(F)
  const F4_VS_F3 = F.difference(futilityMethods, R.keys(OLD))
  expect(F4_VS_F3).toMatchSnapshot()
})
