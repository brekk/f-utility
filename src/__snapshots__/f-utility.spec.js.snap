// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`methods 1`] = `
Array [
  "$",
  "C",
  "F",
  "I",
  "K",
  "PLACEHOLDER",
  "T",
  "__",
  "add",
  "addIndex",
  "all",
  "always",
  "and",
  "any",
  "anyPass",
  "ap",
  "binarySideEffect",
  "both",
  "box",
  "complement",
  "concat",
  "cond",
  "constant",
  "curry",
  "curryN",
  "custom",
  "def",
  "difference",
  "divide",
  "either",
  "entries",
  "equals",
  "every",
  "filter",
  "first",
  "flip",
  "forEach",
  "freeze",
  "fromEntries",
  "fromPairs",
  "gt",
  "gte",
  "identity",
  "ifElse",
  "includes",
  "inspect",
  "is",
  "isArray",
  "isBoolean",
  "isFunction",
  "isNumber",
  "isObject",
  "isRawObject",
  "isString",
  "isSymbol",
  "isUndefined",
  "isUnmatched",
  "j2",
  "jam",
  "join",
  "keys",
  "last",
  "length",
  "lt",
  "lte",
  "map",
  "mash",
  "memoizeWith",
  "merge",
  "mergeRight",
  "multiply",
  "not",
  "nth",
  "or",
  "path",
  "pathEq",
  "pathIs",
  "pathOr",
  "pathSatisfies",
  "pipe",
  "prop",
  "propEq",
  "propIs",
  "propOr",
  "propSatisfies",
  "range",
  "reduce",
  "reject",
  "sideEffect",
  "sideEffect2",
  "slice",
  "smooth",
  "some",
  "sort",
  "split",
  "subtract",
  "temper",
  "toJSON",
  "toLower",
  "toPairs",
  "toUpper",
  "trace",
  "uniq",
  "version",
  "when",
]
`;

exports[`methods 2`] = `
Array [
  "$",
  "C",
  "I",
  "K",
  "PLACEHOLDER",
  "binarySideEffect",
  "box",
  "constant",
  "custom",
  "def",
  "entries",
  "every",
  "first",
  "freeze",
  "fromEntries",
  "inspect",
  "isArray",
  "isBoolean",
  "isFunction",
  "isNumber",
  "isObject",
  "isRawObject",
  "isString",
  "isSymbol",
  "isUndefined",
  "isUnmatched",
  "j2",
  "jam",
  "mash",
  "pathIs",
  "sideEffect",
  "sideEffect2",
  "smooth",
  "some",
  "temper",
  "toJSON",
  "trace",
  "version",
]
`;

exports[`methods 3`] = `
Array [
  "adjust",
  "allPass",
  "aperture",
  "append",
  "apply",
  "applySpec",
  "applyTo",
  "ascend",
  "assoc",
  "assocPath",
  "binary",
  "bind",
  "call",
  "chain",
  "clamp",
  "clone",
  "comparator",
  "compose",
  "composeK",
  "composeP",
  "composeWith",
  "construct",
  "constructN",
  "contains",
  "converge",
  "countBy",
  "dec",
  "defaultTo",
  "descend",
  "differenceWith",
  "dissoc",
  "dissocPath",
  "drop",
  "dropLast",
  "dropLastWhile",
  "dropRepeats",
  "dropRepeatsWith",
  "dropWhile",
  "empty",
  "endsWith",
  "eqBy",
  "eqProps",
  "evolve",
  "find",
  "findIndex",
  "findLast",
  "findLastIndex",
  "flatten",
  "forEachObjIndexed",
  "groupBy",
  "groupWith",
  "has",
  "hasIn",
  "hasPath",
  "head",
  "identical",
  "inc",
  "indexBy",
  "indexOf",
  "init",
  "innerJoin",
  "insert",
  "insertAll",
  "intersection",
  "intersperse",
  "into",
  "invert",
  "invertObj",
  "invoker",
  "isEmpty",
  "isNil",
  "juxt",
  "keysIn",
  "lastIndexOf",
  "lens",
  "lensIndex",
  "lensPath",
  "lensProp",
  "lift",
  "liftN",
  "mapAccum",
  "mapAccumRight",
  "mapObjIndexed",
  "match",
  "mathMod",
  "max",
  "maxBy",
  "mean",
  "median",
  "mergeAll",
  "mergeDeepLeft",
  "mergeDeepRight",
  "mergeDeepWith",
  "mergeDeepWithKey",
  "mergeLeft",
  "mergeWith",
  "mergeWithKey",
  "min",
  "minBy",
  "modulo",
  "move",
  "nAry",
  "negate",
  "none",
  "nthArg",
  "o",
  "objOf",
  "of",
  "omit",
  "once",
  "otherwise",
  "over",
  "pair",
  "partial",
  "partialRight",
  "partition",
  "pick",
  "pickAll",
  "pickBy",
  "pipeK",
  "pipeP",
  "pipeWith",
  "pluck",
  "prepend",
  "product",
  "project",
  "props",
  "reduceBy",
  "reduceRight",
  "reduceWhile",
  "reduced",
  "remove",
  "repeat",
  "replace",
  "reverse",
  "scan",
  "sequence",
  "set",
  "sortBy",
  "sortWith",
  "splitAt",
  "splitEvery",
  "splitWhen",
  "startsWith",
  "sum",
  "symmetricDifference",
  "symmetricDifferenceWith",
  "tail",
  "take",
  "takeLast",
  "takeLastWhile",
  "takeWhile",
  "tap",
  "test",
  "then",
  "times",
  "toPairsIn",
  "toString",
  "transduce",
  "transpose",
  "traverse",
  "trim",
  "tryCatch",
  "type",
  "unapply",
  "unary",
  "uncurryN",
  "unfold",
  "union",
  "unionWith",
  "uniqBy",
  "uniqWith",
  "unless",
  "unnest",
  "until",
  "update",
  "useWith",
  "values",
  "valuesIn",
  "view",
  "where",
  "whereEq",
  "without",
  "xprod",
  "zip",
  "zipObj",
  "zipWith",
  "thunkify",
]
`;

exports[`methods 4`] = `
Array [
  "curryify",
  "curryObjectK",
  "curryObjectN",
  "curryObjectKN",
  "remap",
  "remapArray",
  "compose",
  "random",
  "symmetricDifference",
  "alterIndex",
  "alterFirstIndex",
  "alterLastIndex",
  "relativeIndex",
  "choice",
  "fork",
  "iterate",
  "fold",
  "chain",
  "flatMap",
  "equal",
  "round",
  "pow",
  "invert",
  "charAt",
  "codePointAt",
  "endsWith",
  "indexOf",
  "lastIndexOf",
  "match",
  "padEnd",
  "padStart",
  "repeat",
  "replace",
  "search",
  "startsWith",
  "substr",
  "trim",
  "ternary",
  "triplet",
  "assign",
  "mapTuple",
  "mapTuples",
  "mapKeys",
  "pairwise",
  "pairwiseObject",
  "isTypeof",
  "isNil",
  "isDistinctObject",
  "isPOJO",
  "objectLength",
  "anyLength",
  "which",
]
`;

exports[`methods 5`] = `
Array [
  "C",
  "F",
  "T",
  "__",
  "addIndex",
  "all",
  "always",
  "and",
  "any",
  "anyPass",
  "binarySideEffect",
  "both",
  "box",
  "complement",
  "cond",
  "constant",
  "curryN",
  "custom",
  "def",
  "either",
  "first",
  "forEach",
  "fromEntries",
  "gt",
  "gte",
  "identity",
  "ifElse",
  "includes",
  "inspect",
  "is",
  "isRawObject",
  "isSymbol",
  "isUndefined",
  "isUnmatched",
  "j2",
  "jam",
  "last",
  "lt",
  "lte",
  "mash",
  "memoizeWith",
  "mergeRight",
  "nth",
  "or",
  "pathIs",
  "sideEffect",
  "sideEffect2",
  "slice",
  "smooth",
  "temper",
  "toJSON",
  "toLower",
  "toUpper",
  "trace",
  "uniq",
  "when",
]
`;
